src/Microsoft.TemplateEngine.Edge/Settings/TemplateCache.cs

```diff
@@ -13,193 +15,116 @@ namespace Microsoft.TemplateEngine.Edge.Settings
 {
     internal class TemplateCache
     {
-        // locale -> identity -> locator
-        private readonly IDictionary<string, IDictionary<string, ILocalizationLocator>> _localizationMemoryCache = new Dictionary<string, IDictionary<string, ILocalizationLocator>>();
-
-        private readonly IEngineEnvironmentSettings _environmentSettings;
-        private readonly Paths _paths;
-        private IDictionary<string, ITemplate> _templateMemoryCache = new Dictionary<string, ITemplate>();
-        private Scanner _installScanner;
-
-        public TemplateCache(IEngineEnvironmentSettings environmentSettings)
-        {
-            _environmentSettings = environmentSettings;
-            _paths = new Paths(environmentSettings);
-            TemplateInfo = new List<TemplateInfo>();
-            MountPointsInfo = new Dictionary<string, DateTime>();
-            Locale = CultureInfo.CurrentUICulture.Name;
-        }
-
-        public TemplateCache(IEngineEnvironmentSettings environmentSettings, JObject parsed)
-            : this(environmentSettings)
+        public TemplateCache(IEnumerable<ScanResult> scanResults, Dictionary<string, DateTime> mountPoints)
         {
-            ParseCacheContent(parsed);
-        }
-
-        [JsonProperty]
-        public string Version { get; private set; }
-
-        [JsonProperty]
-        public string Locale { get; private set; }
+            var localizationMemoryCache = new Dictionary<string, IDictionary<string, ILocalizationLocator>>();
+            var templateMemoryCache = new Dictionary<string, ITemplate>();
 
-        [JsonProperty]
-        public IReadOnlyList<TemplateInfo> TemplateInfo { get; set; }
-
-        [JsonProperty]
-        public Dictionary<string, DateTime> MountPointsInfo { get; set; }
-
-        private Scanner InstallScanner
-        {
-            get
+            foreach (var scanResult in scanResults)
             {
-                if (_installScanner == null)
+                foreach (ILocalizationLocator locator in scanResult.Localizations)
                 {
-                    _installScanner = new Scanner(_environmentSettings);
+                    if (!localizationMemoryCache.TryGetValue(locator.Locale, out IDictionary<string, ILocalizationLocator> localeLocators))
```

Comment:
Ouch. I'm seeing some conflicts with my PR. Never mind 😄

Reply:
We can hop on call and resolve conflicts together if it makes it easier 😉